### PR TITLE
Fix disabling "download" buttons after changing route

### LINKED ISSUE #364: Going to Print Mode and back after compiling disables firmware and source download buttons.
Steps to reproduce:

1. Create a keymap and compile it. Wait for the compile task to complete. Don't download anything yet.
2. Click `Print Keymap`.
3. Click `< Back`.

The `Keymap Only`, `Full Source`, and `Firmware` download buttons will all be disabled.

### PR DESCRIPTION
**Pull request by mateossh**
## Description

Keyboard in store is updated on every route visit. `disableOtherButtons` was executed every time without any conditions in `postUpdateKeyboard`.
I fixed bug by moving `disableOtherButtons` to `updateKeyboard` and checking if "new" keyboard is different than previous one.

Also I want to point out this was my first time writing tests. If you have any ideas how can I make them better, let me know.

Fixed #364

### CODE REVIEW

**Review by noroadsleft — APPROVED** [review]
Confirm this fixes the linked issue, but I know nothing about unit tests.

File: tests/integration/fix-364.spec.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    cy.intercept('POST', 'https://api.qmk.fm/v1/compile', { fixture: '364-first-step.json' });
+    cy.intercept('GET', 'https://api.qmk.fm/v1/compile/ea1514b3-bdfc-4a7b-9b5c-08752684f7f6', 
+      { fixture: '364-second-step.json' });
+  });
+
+  it('"download firmware" button should be enabled after pressing compile', function () {
+    cy.wait(1000);
+
+    cy.get('#compile', { timeout: 5000 }).click();
```
Comment (yanfali) [inline tests/integration/fix-364.spec.js:19]: does this test even work? compiles can take minutes.

Comment (mateossh) [reply tests/integration/fix-364.spec.js:19]: This test uses feature called 'fixtures' (API mocking). 

When configurator sends request POST `https://api.qmk.fm/v1/compile` cypress sends response from `364-first-step.json` and when applications sends GET `https://api.qmk.fm/v1/compile/{id}` then cypress sends response from `364-second-step.json`.

File: tests/fixtures/364-second-step.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+         "command": [
+           "qmk",
+           "compile",
+           "newgame40-layout_ortho_4x10_mine.json"
+         ],
+         "id": "ca230f62-b4e6-4647-8fc0-0c3879ab0d5e",
+         "firmware_binary_url": [
+           "https:api.qmk.fm/v1/compile/ca230f62-b4e6-4647-8fc0-0c3879ab0d5e/download"
```
Comment (yanfali) [inline tests/fixtures/364-second-step.json:26]: these aren't valid urls

File: tests/integration/fix-364.spec.js (hunk 1 above)
Comment (yanfali) [reply tests/integration/fix-364.spec.js:19]: I see. thanks for clarifying. It's not a feature I've used. Very neat

**Review by yanfali — APPROVED** [review]

File: tests/fixtures/364-second-step.json (hunk 2 above)
Comment (yanfali) [reply tests/fixtures/364-second-step.json:26]: fix up the missing :// and we'll merge

Comment (mateossh) [reply tests/fixtures/364-second-step.json]: Fixed

Comment (yanfali) [reply tests/fixtures/364-second-step.json]: Thank you!

**Pull request merged**